drivers/soundwire/bus_type.c

```diff
@@ -23,10 +23,16 @@ sdw_get_device_id(struct sdw_slave *slave, struct sdw_driver *drv)
 	const struct sdw_device_id *id = drv->id_table;
 
 	while (id && id->mfg_id) {
-		if (slave->id.mfg_id == id->mfg_id &&
-		    slave->id.part_id == id->part_id)
+		if ((id->sdw_version &&
+		     slave->id.sdw_version != id->sdw_version) ||
+		    (id->class_id &&
+		     slave->id.class_id != id->class_id) ||
+		    slave->id.mfg_id != id->mfg_id ||
+		    slave->id.part_id != id->part_id) {
+			id++;
+		} else {
```

Comment:
so the whole could be
```
for (id = drv->id_table; id && id->mfg_id; id++)
	if ((!id->sdw_version || slave->id.sdw_version == id->sdw_version) &&
	    (!id->class_id || slave->id.class_id == id->class_id) &&
	    slave->id.mfg_id == id->mfg_id &&
	    slave->id.part_id == id->part_id)
		return id;
```

Reply:
looks much better indeed, will follow your suggestion,

Reply:
fixed now